cmd/crowdsec-cli/cliallowlists/allowlists.go

```diff
@@ -0,0 +1,562 @@
+package cliallowlists
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/crowdsecurity/crowdsec/cmd/crowdsec-cli/cstable"
+	"github.com/crowdsecurity/crowdsec/cmd/crowdsec-cli/require"
+	"github.com/crowdsecurity/crowdsec/pkg/apiclient"
+	"github.com/crowdsecurity/crowdsec/pkg/csconfig"
+	"github.com/crowdsecurity/crowdsec/pkg/database"
+	"github.com/crowdsecurity/crowdsec/pkg/models"
+	"github.com/crowdsecurity/go-cs-lib/cstime"
+	"github.com/fatih/color"
+	"github.com/go-openapi/strfmt"
+	"github.com/jedib0t/go-pretty/v6/table"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type configGetter = func() *csconfig.Config
+
+type cliAllowLists struct {
+	db     *database.Client
+	client *apiclient.ApiClient
+	cfg    configGetter
+}
+
+func New(cfg configGetter) *cliAllowLists {
+	return &cliAllowLists{
+		cfg: cfg,
+	}
+}
+
+// validAllowlists returns a list of valid allowlists name for command completion
+func (cli *cliAllowLists) validAllowlists(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	var err error
+
+	cfg := cli.cfg()
+	ctx := cmd.Context()
+
+	// need to load config and db because PersistentPreRunE is not called for completions
+
+	if err = require.LAPI(cfg); err != nil {
+		cobra.CompError("unable to load LAPI " + err.Error())
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	cli.db, err = require.DBClient(ctx, cfg.DbConfig)
+	if err != nil {
+		cobra.CompError("unable to load dbclient " + err.Error())
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	allowlists, err := cli.db.ListAllowLists(ctx, false)
+	if err != nil {
+		cobra.CompError("unable to list allowlists " + err.Error())
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	ret := []string{}
+
+	for _, allowlist := range allowlists {
+		if strings.Contains(allowlist.Name, toComplete) && !slices.Contains(args, allowlist.Name) && !allowlist.FromConsole {
+			ret = append(ret, allowlist.Name)
+		}
+	}
+
+	return ret, cobra.ShellCompDirectiveNoFileComp
+}
+
+func (cli *cliAllowLists) validAllowlistsWithConsole(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

Comment:
comment on this function purpose and the why (ie. allowlists from console and cscli are strictly separated).